state.go

```diff
@@ -226,6 +235,9 @@ type State interface {
 	// Return transition
 	Transition() Transition
 	SetTransition(t Transition)
+	// SaveRecoveryData stores the data needed to recreate the state
+	// in case of powerloss
+	SaveRecoveryData(lt Transition, s store.Store)
```

Comment:
I am wondering here. Now all the states need to implement `SaveRecoveryData`. Wouldn't it be easier to store the data for all of them by default and ignore the ones we don't need? One way could be to add an extra field to `baseState` (`storeRecoveryData bool`) or similar.